examples/meson.build

```diff
@@ -0,0 +1,14 @@
+if not get_option('examples').enabled()
+    subdir_done()
+endif
```

Comment:
The usual pattern/convention is to do `if not get_option(..).disabled()` so that it's also built with auto, but not a big deal of course.

Reply:
(updated to remove extraneous `not`, sorry)

Reply:
Ok I'll update it that way. I wasn't sure which way was idiomatic.

It's a bit unclear what "auto" means in this context to be honest, is the point to delegate that to the packager so that they chose if all automatic feature should be enabled/disabled all at once?

Reply:
Updated with `if foo.disabled()`

Reply:
Right so packagers/integrators can easily override it one way or another that way.

This might make more sense when there are external dependencies involved, e.g. one can do `optional_dep = dependency('liboptional', required: get_option('foo'))` and if the option is `enabled` that means it will error out if it's not present, if `disabled` it will skip the check and return not-found and if `auto` it will check and either return the lib or not-found.

With things like tests/examples there's no check of course but it's still nice to have it as a feature option so it can be flipped one way or another using the global `--auto-features=...` switch.

Developers building from git/tarball will not override it typically, in which case it should just build it if there are no external deps imho.

Matter of opinion I guess, but that's what most projects do afaik :)